`SerializeComponents::serialize_recursive` can't infer type for `MS`

## Description

I'm trying to use `SerializeComponents::serialize_recursive` pretty much using the code from the [saveload example](https://github.com/slide-rs/specs/blob/master/examples/saveload.rs).

## Meta

Rust version: rustc 1.36.0-nightly (8dd4aae9a 2019-05-04)
Specs version / commit: 0.14.3
Operating system: Windows 10

## Reproduction

Change the system slightly:
```rust
    struct Serialize;
    impl<'a> System<'a> for Serialize {
        type SystemData = (
            Entities<'a>,
            ReadStorage<'a, Pos>,
            ReadStorage<'a, Mass>,
            WriteStorage<'a, U64Marker>,
            Write<'a, U64MarkerAllocator>,
        );

        fn run(&mut self, (ents, pos, mass, markers, marker_allocator): Self::SystemData) {
            let mut ser = ron::ser::Serializer::new(Some(Default::default()), true);
            SerializeComponents::<NoError, U64Marker>::serialize_recursive(
                &(&pos, &mass),
                &ents,
                &mut markers,
                &mut marker_allocator,
                &mut ser,
            )
            .unwrap_or_else(|e| eprintln!("Error: {}", e));
            println!("{}", ser.into_output_string());
        }
    }
```

```
error[E0282]: type annotations needed
   --> examples\saveload.rs:173:13
    |
173 |             SerializeComponents::<NoError, U64Marker>::serialize_recursive(
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for `MS`
```

## Expected behavior

Compiles and works.

--- a/src/saveload/ser.rs
+++ b/src/saveload/ser.rs
@@ -63,7 +63,7 @@ where
     /// in `ids` argument marks unmarked `Entity` (the marker of which was
     /// requested) and it will get serialized recursively.
     /// For serializing without such recursion see `serialize` function.
-    fn serialize_recursive<MS, S>(
+    fn serialize_recursive<S>(
         &self,
         entities: &EntitiesRes,
         markers: &mut WriteStorage<M>,

